vdr/vdr.go

```diff
@@ -68,6 +70,37 @@ type VDR struct {
 	eventManager      events.Event
 }
 
+func (r *VDR) DeriveWebDID(ctx context.Context, baseURL url.URL, nutsDID did.DID) (*did.Document, error) {
+	webDID, err := didweb.URLToDID(*baseURL.JoinPath(nutsDID.ID))
+	if err != nil {
+		return nil, fmt.Errorf("unable to derive Web DID from Nuts DID (%s): %w", nutsDID, err)
+	}
+
+	nutsDIDDocument, _, err := r.Resolver().Resolve(nutsDID, nil)
+	if err != nil {
+		return nil, err
+	}
+	isOwner, err := r.IsOwner(ctx, nutsDID)
+	if err != nil {
+		return nil, err
+	}
+	if isOwner {
+		nutsDIDDocumentData, _ := nutsDIDDocument.MarshalJSON()
+		webDIDDocumentData := bytes.ReplaceAll(nutsDIDDocumentData, []byte(nutsDID.String()), []byte(webDID.String()))
```

Comment:
Does this also rewrites the document controller?

Reply:
It only rewrites the controller if the document is its own controller. But question is what value the controller has for `did:web`,  since `did:nuts` uses it for authentication DID document updates, which we don't have for `did:web`. The DID spec says for controller:

> When a [controller](https://www.w3.org/TR/did-core/#dfn-controller) property is present in a [DID document](https://www.w3.org/TR/did-core/#dfn-did-documents), its value expresses one or more [DIDs](https://www.w3.org/TR/did-core/#dfn-decentralized-identifiers). Any [verification methods](https://www.w3.org/TR/did-core/#dfn-verification-method) contained in the [DID documents](https://www.w3.org/TR/did-core/#dfn-did-documents) for those [DIDs](https://www.w3.org/TR/did-core/#dfn-decentralized-identifiers) SHOULD be accepted as authoritative, such that proofs that satisfy those [verification methods](https://www.w3.org/TR/did-core/#dfn-verification-method) are to be considered equivalent to proofs provided by the [DID subject](https://www.w3.org/TR/did-core/#dfn-did-subjects).

and:

> NOTE: Authorization vs authentication
Note that authorization provided by the value of controller is separate from authentication as described in [5.3.1 Authentication](https://www.w3.org/TR/did-core/#authentication). This is particularly important for key recovery in the case of cryptographic key loss, where the [DID subject](https://www.w3.org/TR/did-core/#dfn-did-subjects) no longer has access to their keys, or key compromise, where the [DID controller](https://www.w3.org/TR/did-core/#dfn-did-controllers)'s trusted third parties need to override malicious activity by an attacker. See [9. Security Considerations](https://www.w3.org/TR/did-core/#security-considerations) for information related to threat models and attack vectors.

I think we can probably leave it as it currently is?